lib/wikidata/diff/analyzer.rb

```diff
@@ -46,6 +51,77 @@ def self.get_revision_content(revision_id)
   end
 
 
+  def self.isolate_claim_differences(current_content, parent_content)
```

Comment:
I wonder if the `id` and `hash` values from the JSON might be useful here. I don't know for sure that it works like this, but I would expect it to work like this: any time a claim is added, it will be represented by a claim object in the JSON with an `id` that wasn't in the parent, and any time a claim is changed, it will result in a different `hash` for a claim but not a change in `id`.

Reply:
I just checked https://www.wikidata.org/w/index.php?title=Q111269579&diff=1898167685&oldid=1898167431 (Changed Claim by adding a reference). the hash (inside mainsnak) stays the same in both revision - just the reference gets added in the new claim.